ehealthid-rp/src/test/java/com/oviva/ehealthid/relyingparty/ws/ui/PagesTest.java

```diff
@@ -21,17 +29,127 @@ void selectIdpForm() {
                 new IdpEntry("https://a.example.com", "AoK Tesfalen", null),
                 new IdpEntry("https://b.example.com", "Siemens", null),
                 new IdpEntry("https://c.example.com", "Zuse", null),
-                new IdpEntry("https://d.example.com", "Barmer", null)));
+                new IdpEntry("https://d.example.com", "Barmer", null)),
+            "de-DE;q=0.1,en-US;q=0.5");
 
     assertEquals(Fixtures.getUtf8String("pages_golden_idp-select-form.bin"), rendered);
   }
 
+  @ParameterizedTest
+  @MethodSource("provideLanguageAndMessageForSelectIdp")
+  void selectIdpForm_withInternationalisation(String acceptLanguageHeaderValue, String message) {
+    var sut = new Pages(renderer);
+
+    var rendered =
+        sut.selectIdpForm(
+            List.of(
+                new IdpEntry("https://a.example.com", "AoK Tesfalen", null),
+                new IdpEntry("https://b.example.com", "Siemens", null),
+                new IdpEntry("https://c.example.com", "Zuse", null),
+                new IdpEntry("https://d.example.com", "Barmer", null)),
+            acceptLanguageHeaderValue);
+
+    assertTrue(rendered.contains(message));
+  }
+
   @Test
-  void error() {
+  void error_withFixture() {
     var sut = new Pages(renderer);
 
-    var rendered = sut.error("Ohhh noes!");
+    var rendered = sut.error(new Message("error.serverError", ""), "de-DE;q=0.1,en-US;q=0.5");
 
     assertEquals(Fixtures.getUtf8String("pages_golden_error.bin"), rendered);
   }
+
+  @ParameterizedTest
+  @MethodSource("provideKeyAndMessageError")
+  void error_simpleError(String acceptLanguageHeaderValue, String messageKey, String message) {
+    var sut = new Pages(renderer);
+
+    var rendered = sut.error(new Message(messageKey, ""), acceptLanguageHeaderValue);
+
+    assertTrue(rendered.contains(message));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideKeyWithDynamicContentMessageError")
+  void error_errorWithDynamicContent(
+      String acceptLanguageHeaderValue, String messageKey, String message) {
+    var sut = new Pages(renderer);
+    var uri = URI.create("https://idp.example.com");
+
+    var rendered =
+        sut.error(new Message(messageKey, String.valueOf(uri)), acceptLanguageHeaderValue);
+    assertTrue(StringEscapeUtils.unescapeHtml4(rendered).contains(message));
+  }
+
+  private static Stream<Arguments> provideLanguageAndMessageForSelectIdp() {
+    return Stream.of(
+        Arguments.of("de-DE;q=0.4,en-US;q=0.8", "en-US"),
+        Arguments.of("de-DE,en-US;q=0.4", "de-DE"),
+        Arguments.of("de-DE;q=0.4,en-US;q=0.8,it,ch-DE", "Login with GesundheitsID"),
+        Arguments.of("de-DE;q=0.9,en-US;q=0.8,it,jp", "Anmeldung mit GesundheitsID"),
+        Arguments.of("en-US", "Select your GesundheitsID Provider"),
+        Arguments.of("de-DE", "Wählen Sie Ihren GesundheitsID Anbieter"),
+        Arguments.of("en-US,de-CH", "Login"),
```

Comment:
```suggestion
        Arguments.of("en-US,de-CH", "Login"),
        Arguments.of("de-CH,en-US", "Einloggen"),
```
Does fallback work on the language level?